File: 00167.py
class Solution:
    def twoSum2(self, numbers: list[int], target: int) -> list[int]:
        left = 0
        right = len(numbers) - 1

        while left < right:
            s = numbers[left] + numbers[right]
            if s == target:
                return [left + 1, right + 1]
            elif s > target:
                right -= 1
            else:
                left += 1
        return



def main():
    s = Solution()
    numbers = [2,7,11,15] 
    target = 9
    print(s.twoSum(numbers, target))

    numbers = [2,3,4] 
    target = 6
    print(s.twoSum(numbers, target))

    numbers = [-1,0] 
    target = -1
    print(s.twoSum(numbers, target))

    return



if __name__ == '__main__':
    main()

File: 00167.c
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize)
{
    int left = 0;
    int right = numbersSize - 1;


    while (left < right) {
        if (numbers[left] + numbers[right] == target) {
            break;
        } else if (numbers[left] + numbers[right] < target) {
            left++;
        } else {
            right--;
        }
    }

    *returnSize = 2;
    int *ret = (int*)malloc(sizeof(int) * 2);

    ret[0] = left + 1;
    ret[1] = right + 1;
    return ret;
}
